models/Foundation.js

```diff
@@ -0,0 +1,63 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+const foundationSchema = mongoose.Schema({
+  email: {
+    type: String,
+    match: /.+\@.+\..+/,
+    required: [true, "Email is required"],
+    validate: {
+      validator: async function (value) {
+        const foundation = await Foundation.findOne({ email: value });

```

Comment:
If users and foundations cannot sign up with same email, you need to add an extra validation